src/MonoTorrent.Client/MonoTorrent.Client/PeerId.PieceRequester.cs

```diff
@@ -61,6 +61,7 @@ public partial class PeerId : IRequester
 
         int IRequester.PreferredRequestAmount (int pieceLength)
         {
+            return 1;
```

Comment:
This made it easier to trigger the bug. 8 pieces were being requested instead of 1, which was weird.